### PR TITLE
Add sumologic_source provider: issue #43

### PR DESCRIPTION
**Pull request by kevindickerson**
Addresses Issue #43.
---

Adds new provider, `sumologic_source`, with actions `:create` and `:delete`.

- `:create` simply drops a template given a set of parameters. This puts the onus of JSON creation back on the consumer cookbook, where it belongs. The consumer cookbook should be responsible for generating, writing, and verifying the JSON as a valid configuration for the service. (I believe SumoLogic fails silently when a configuration file is bad.) There is room for debate on where the validation should occur.
- When we create a source, we recursively create the directory that it's in, but when we delete a source, we don't delete the directory. A probably good change may be to fail if the directory doesn't exist, and then leave it up to the consumer to create the directory.
- I did not add test suites for Windows! If this is desired, then I need to have a discussion about the current test suites, which don't converge for me on the Windows platform.
- TL;DR This is a pretty dumb provider that just drops a template. I've allowed the consumer to pass the `cookbook` value into the provider to kick the ball down the road as far as templates are concerned.

@RoboticCheese @lmickh @kennonkwok

### CODE REVIEW

**Comment by kevindickerson** [comment]
(Comment moved from issues)

- Being not an expert at Sumologic, I'm trying to find out where a JSON 'source' file should actually go... Any ideas?

- And also, just to confirm, the sumo.conf appears to me to be configurable to set like `sources=somedir`, and then just drop a json file at that specified location. Does that sound sane?

**Comment by lmickh** [comment]
Believe you have to set the syncsources option as well.  Otherwise it only consumes the json at install.   The sumologic guys could verify this.

**Comment by duchatran** [comment]
@kevindickerson, @lmickh , first off thanks much for the pull request. I like the idea, but looks like you guys are using LWRP. Is there any plan to change to the new custom resource model? I am not sure if it makes sense to merge something that is going to be obsoleted soon. Let me know. I'll be merging another request that use the new model, so maybe you guys can start from there?

**Comment by duchatran** [comment]
Latest master branch has incorporated a more modern approach for custom provider, so I'll close this request.
